internal/internal.go

```diff
@@ -498,11 +503,106 @@ func (i *Internal) getIDFromHost(host string) (string, error) {
 	return "", fmt.Errorf("no ingress found for host %s", host)
 }
 
-// VICEStatus handles requests to check the status of a running VICE app in K8s.
+// URLReadyHandler returns whether or not a VICE app is ready
+// for users to access it. This version will check the user's permissions
+// and return an error if they aren't allowed to access the running app.
+func (i *Internal) URLReadyHandler(c echo.Context) error {
+	var (
+		ingressExists bool
+		serviceExists bool
+		podReady      bool
+	)
+
+	user := c.QueryParam("user")
+	if user == "" {
+		return echo.NewHTTPError(http.StatusBadRequest, "user query parameter must be set")
+	}
+
+	// Since some usernames don't come through the labelling process unscathed, we have to use
+	// the user ID.
+	fixedUser := i.fixUsername(user)
+	a := apps.NewApps(i.db, i.UserSuffix)
+	_, err := a.GetUserID(fixedUser)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return echo.NewHTTPError(http.StatusNotFound, fmt.Sprintf("user %s not found", fixedUser))
+		}
+		return echo.NewHTTPError(http.StatusInternalServerError, err.Error())
+	}
+
+	host := c.Param("host")
+
+	// Use the name of the ingress to retrieve the externalID
+	id, err := i.getIDFromHost(host)
+	if err != nil {
+		return echo.NewHTTPError(http.StatusNotFound, err.Error())
+	}
```

Comment:
I could be wrong, but it seems to me that it's possible that this code could cause the endpoint to return a 404 in cases where we're unable to connect to the Kubernetes API.

Reply:
Yeah, that's annoying. I'll change it to return a 500, which is more accurate in this instance anyway.